test/widgets/message_list_test.dart

```diff
@@ -981,4 +981,70 @@ void main() {
       });
     });
   });
+
+  group('TypingStatus', () {
+    final users = [eg.selfUser, eg.otherUser, eg.thirdUser, eg.fourthUser];
+    final finder = find.descendant(
+      of: find.byType(TypingStatusWidget),
+      matching: find.byType(Text)
+    );
+
+    checkTyping(WidgetTester tester, TypingEvent event, {required String expected}) async {
```

Comment:
nit: function definition should have explicit return type, so it doesn't look like a function call

(hmm, we should really look to see if there's a lint rule for that, and turn it on — seems eminently lintable)